core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/FcJsonRpc.java

```diff
@@ -98,6 +102,29 @@ public boolean filecoinWalletHas(@JsonRpcParam("address") final String address)
     return fcSigners.availableIdentifiers().contains(address);
   }
 
+  @JsonRpcMethod("Filecoin.WalletSignMessage")
+  public FilecoinSignedMessage filecoinSignMessage(
+      @JsonRpcParam("Version") final Integer version,
+      @JsonRpcParam("To") final String to,
+      @JsonRpcParam("From") final String from,
+      @JsonRpcParam("Nonce") final Integer nonce,
+      @JsonRpcParam("Value") final String value,
+      @JsonRpcParam("GasPrice") final String gasPrice,
+      @JsonRpcParam("GasLimit") final Integer gasLimit,
+      @JsonRpcParam("Method") final Integer method,
+      @JsonRpcParam("Params") final String params) {
+    final FilecoinMessage message =
+        new FilecoinMessage(version, to, from, nonce, value, gasPrice, gasLimit, method, params);
+    final ObjectMapper cborObjectMapper = new ObjectMapper(new CBORFactory());
+    try {
+      final byte[] bytes = cborObjectMapper.writeValueAsBytes(message);
+      return new FilecoinSignedMessage(
+          message, filecoinWalletSign(from, Bytes.wrap(bytes).toBase64String()));
+    } catch (JsonProcessingException e) {
```

Comment:
nit: `final JsonProcessingException`

Reply:
so, this was the thing you were talking about ... we are sending back `message` and signed `message` as well.

Reply:
done